### ISSUE TITLE
UI: Inconsistent behaviour / permission denied with "App in any namespace"

### ISSUE THREAD

**Issue by woehrl01** [argoproj/argo-cd#17781]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Navigating between clusters across different namespaces, it often results in "permission denied" errors. Navigating back to "Applications" and then accessing the app works.

Additionally the namespace is correctly prefixed, in the "Application list". In the "Application View" the search is missing the namespace, making the selection harder.

**To Reproduce**

Switch to an application inside the "argocd" namespace. Switch to an application via the search, to a different namespace.

**Expected behavior**

The search in the UI contains the same structure of application names and navigating between the views works correctly without any permission errors.

**Screenshots**

![image](https://github.com/argoproj/argo-cd/assets/2535856/ee8bd6ab-c6e4-4bb8-aa57-7e1d446fd69a)

**Version**

```shell
Paste the output from `argocd version` here.
v2.10.6+d504d2b
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by Sir-Jacques** [comment]
I'm experiencing the same with v2.10.2. It seems to have to do with the url during navigation (changes if argo-applications are in different namespaces).

This was my observation of this issue:

I have argo-applications that are in the "argocd" namespace. They will have UI url `/applications/argocd/xxxxx`. Another app is in namespace "argoapps", they have url `/applications/argoapps/xxxxx`. When on the first argo-application page in the UI, and navigating to the other argo-application (in another namespace) using the searchbar on top, it navigates to `/applications/xxxxx` (without a namespace). When I add the namespace manually, the page opens correctly.

Not sure, but this seems to have to do with the old way the urls were constructed? Might the old urls still be in use in some navigation function?

**Comment by Sir-Jacques** [comment]
I've experimented a bit and found a workaround (and likely the underlying cause). The application searchbar (on top of the application-details page) did not construct the correct URLs during navigation. 

On top of that the ApplicationDetails class does not seem to run its constructor again when nagivating. That is causing issues as the appNamespace is read/updated in this constructor. By using the props element directly I got it to nagivate nicely without "permission denied" errors.

https://github.com/argoproj/argo-cd/pull/17800
